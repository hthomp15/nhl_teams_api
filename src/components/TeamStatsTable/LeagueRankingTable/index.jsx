import React from "react";


function LeagueRankingTable(props) {
    const { leagueRanking } = props;

    return (
        <table className="league-rankings-table table-auto w-3/4 border-2 border-collapse mb-4">
        <caption className="font-bold text-white bg-blue-200"> Regular Season Rankings </caption>
        <thead>
            <tr>
                <th className="border py-2">Overall</th>
                <th className="border py-2">GPG</th>
                <th className="border py-2">GAPG</th>
                <th className="border py-2">SV%</th>
                <th className="border py-2">S</th>
                <th className="border py-2">SA</th>
                <th className="border py-2">S%</th>
                <th className="border py-2">FOW%</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td className="border px-4 py-2">{leagueRanking.pts}</td>
                <td className="border px-4 py-2">{leagueRanking.goalsPerGame}</td>
                <td className="border px-4 py-2">{leagueRanking.goalsAgainstPerGame}</td>
                <td className="border px-4 py-2">{leagueRanking.savePctRank}</td>
                <td className="border px-4 py-2">{leagueRanking.shotsPerGame}</td>
                <td className="border px-4 py-2">{leagueRanking.shotsAllowed}</td>
                <td className="border px-4 py-2">{leagueRanking.shootingPctRank}</td>
                <td className="border px-4 py-2">{leagueRanking.faceOffWinPercentage}</td>
            </tr>
        </tbody>
    </table>
    )
}

export default LeagueRankingTable;