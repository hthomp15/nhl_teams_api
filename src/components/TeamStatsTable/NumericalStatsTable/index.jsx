import React from "react";

function NumericalStatsTable(props) {
    const { numericalStats } = props;

    const roundNumber = (number) => {
        return Math.round(number * 100) / 100;
    }
    
    return (
        <table className="team-stats-table table-auto w-3/4 border-2 border-collapse mb-4">
        <caption className="font-bold text-white bg-blue-200"> Regular Season Stats </caption>
            <thead className="">
                <tr>
                    <th className="border py-2">GP</th>
                    <th className="border py-2">W</th>
                    <th className="border py-2">L</th>
                    <th className="border py-2">PTS</th>
                    <th className="border py-2">OT</th>
                    <th className="border py-2">GPG</th>
                    <th className="border py-2">GAPG</th>
                    <th className="border py-2">SV%</th>
                    <th className="border py-2">S</th>
                    <th className="border py-2">SA</th>
                    <th className="border py-2">S%</th>
                    <th className="border py-2">FOW%</th>
                </tr>
            </thead>
            <tbody className="text-center">
                <tr>
                    <td className="border px-2 py-2">{numericalStats.gamesPlayed}</td>
                    <td className="border px-2 py-2">{numericalStats.wins}</td>
                    <td className="border px-2 py-2">{numericalStats.losses}</td>
                    <td className="border px-2 py-2">{numericalStats.pts}</td>
                    <td className="border px-2 py-2">{numericalStats.ot}</td>
                    <td className="border px-2 py-2">{roundNumber(numericalStats.goalsPerGame)}</td>
                    <td className="border px-2 py-2">{roundNumber(numericalStats.goalsAgainstPerGame)}</td>
                    <td className="border px-2 py-2">{numericalStats.savePctg}</td>
                    <td className="border px-2 py-2">{roundNumber(numericalStats.shotsPerGame)}</td>
                    <td className="border px-2 py-2">{roundNumber(numericalStats.shotsAllowed)}</td>
                    <td className="border px-2 py-2">{numericalStats.shootingPctg}</td>
                    <td className="border px-2 py-2">{numericalStats.faceOffWinPercentage}</td>
                </tr>
            </tbody>
        </table>
    )
}

export default NumericalStatsTable;