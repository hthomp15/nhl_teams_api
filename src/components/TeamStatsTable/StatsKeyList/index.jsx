import React from "react";

function StatsKeyList() {
    return (
        <table className="table-auto w-1/3 text-center border-2">
            <tbody>
                <tr className="border border-white">
                    <th className="border border-white p-1">GP</th>
                    <td>Games Played</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">W</th>
                    <td>Wins</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">L</th>
                    <td>Losses</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">PTS</th>
                    <td>Points</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">OT</th>
                    <td>Overtime Games</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">GPG</th>
                    <td>Goals Per Game</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">GAPG</th>
                    <td>Goals Against Per Game</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">SV%</th>
                    <td>Save Percentage</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">S</th>
                    <td>Shots</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">SA</th>
                    <td>Shots Allowed</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">S%</th>
                    <td>Shooting Percentage</td>
                </tr>
                <tr className="border border-white">
                    <th className="border border-white p-1">FOW%</th>
                    <td>Faceoff Win Percentage</td>
                </tr>
            </tbody>
        </table>
    );
}

export default StatsKeyList;