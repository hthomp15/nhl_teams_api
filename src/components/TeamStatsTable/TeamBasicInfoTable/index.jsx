import React from "react";
import { useParams } from "react-router-dom";

function TeamBasicInfoTable(props) {
        // Get the correct team from the data with useParams
    const { teamBasicInfo } = props;
    const { id } = useParams();
    // Unable to find the team in the data
    const team = teamBasicInfo.find(function (team, index, array) {
        if (team.id === id) {
            return array;
        }
    }
    );
    console.log("team", team);

    console.log("teamBasicInfo", teamBasicInfo);

    return (
        <table className="team-stats-table table-auto w-3/4 border-2 border-collapse text-center mb-4">
                <thead className="bg-blue-200 text-white borde">
                    <tr>
                        <th className="border ">City</th>
                        <th className="border ">Conference</th>
                        <th className="border ">Division</th>
                        <th className="border ">Venue</th>
                        <th className="border ">Link</th>
                    </tr>
                </thead>
                <tbody className="">
                </tbody>  
            </table>
    );
}

export default TeamBasicInfoTable;