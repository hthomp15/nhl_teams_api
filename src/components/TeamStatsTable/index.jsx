import React from 'react';


function TeamStatsTable(props) {
    const { allStats } = props;
    const { numericalStats, leagueRanking, teamName } = allStats;
    // console.log("numericalStats", numericalStats, "leagueRanking", leagueRanking, "teamName", teamName);
    return (
        <div className="container">
            <h2> {teamName} </h2>
            <table className="team-stats-table table-auto w-full">
            <caption> Regular Season Stats </caption>
                <thead>
                    <tr>
                        <th className="px-4 py-2">GP</th>
                        <th className="px-4 py-2">W</th>
                        <th className="px-4 py-2">L</th>
                        <th className="px-4 py-2">PTS</th>
                        <th className="px-4 py-2">OT</th>
                        <th className="px-4 py-2">GPG</th>
                        <th className="px-4 py-2">GAPG</th>
                        <th className="px-4 py-2">SV%</th>
                        <th className="px-4 py-2">S</th>
                        <th className="px-4 py-2">SA</th>
                        <th className="px-4 py-2">S%</th>
                        <th className="px-4 py-2">FOW%</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td className="border px-4 py-2">{numericalStats.gamesPlayed}</td>
                        <td className="border px-4 py-2">{numericalStats.wins}</td>
                        <td className="border px-4 py-2">{numericalStats.losses}</td>
                        <td className="border px-4 py-2">{numericalStats.pts}</td>
                        <td className="border px-4 py-2">{numericalStats.ot}</td>
                        <td className="border px-4 py-2">{numericalStats.goalsPerGame}</td>
                        <td className="border px-4 py-2">{numericalStats.goalsAgainstPerGame}</td>
                        <td className="border px-4 py-2">{numericalStats.savePctg}</td>
                        <td className="border px-4 py-2">{numericalStats.shotsPerGame}</td>
                        <td className="border px-4 py-2">{numericalStats.shotsAllowed}</td>
                        <td className="border px-4 py-2">{numericalStats.shootingPctg}</td>
                        <td className="border px-4 py-2">{numericalStats.faceOffWinPercentage}</td>
                    </tr>
                </tbody>
            </table>
            <table className="league-rankings-table table-auto w-full">
                <caption> Regular Season Rankings </caption>
                <thead>
                    <tr>
                        <th className="px-4 py-2">Overall</th>
                        <th className="px-4 py-2">GPG</th>
                        <th className="px-4 py-2">GAPG</th>
                        <th className="px-4 py-2">SV%</th>
                        <th className="px-4 py-2">S</th>
                        <th className="px-4 py-2">SA</th>
                        <th className="px-4 py-2">S%</th>
                        <th className="px-4 py-2">FOW%</th>
                    </tr>
                </thead>

                <tbody>

                    <tr>
                        <td className="border px-4 py-2">{leagueRanking.pts}</td>
                        <td className="border px-4 py-2">{leagueRanking.goalsPerGame}</td>
                        <td className="border px-4 py-2">{leagueRanking.goalsAgainstPerGame}</td>
                        <td className="border px-4 py-2">{leagueRanking.savePctRank}</td>
                        <td className="border px-4 py-2">{leagueRanking.shotsPerGame}</td>
                        <td className="border px-4 py-2">{leagueRanking.shotsAllowed}</td>
                        <td className="border px-4 py-2">{leagueRanking.shootingPctRank}</td>
                        <td className="border px-4 py-2">{leagueRanking.faceOffWinPercentage}</td>
                    </tr>
                </tbody>
            </table>
        </div>
    );
}

export default TeamStatsTable;