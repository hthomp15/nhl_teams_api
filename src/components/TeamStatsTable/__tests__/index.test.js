import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import TeamStatsTable from "..";

const numericalStats = { 
    gamesPlayed: "1",
    wins: "1",
    losses: "1",
    pts: "1",
    ot: "1",
    goalsPerGame: "1",
    goalsAgainstPerGame: "1",
    savePctg: "1",
    shotsPerGame: "1",
    shotsAllowed: "1",
    shootingPctg: "20",
    faceOffWinPercentage: "20"
}

const leagueRanking = {
    pts: "1",
    goalsPerGame: "1",
    goalsAgainstPerGame: "1",
    savePctRank: "1",
    shotsPerGame: "1",
    shotsAllowed: "1",
    shootingPctRank: "1",
    faceOffWinPercentage: "1"
}

const teamName = "Test Team";

const allStats = { numericalStats, leagueRanking, teamName }

afterEach(cleanup);

describe("TeamStatsTable Component", () => {
    it("renders", () => {
        render(<TeamStatsTable allStats={allStats} />);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<TeamStatsTable allStats={allStats} />);
        expect(asFragment()).toMatchSnapshot();

    })
});