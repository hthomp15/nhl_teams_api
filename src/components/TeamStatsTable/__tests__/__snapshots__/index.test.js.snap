// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamStatsTable Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
       Test Team 
    </h2>
    <table
      class="team-stats-table table-auto w-full"
    >
      <caption>
         Regular Season Stats 
      </caption>
      <thead>
        <tr>
          <th
            class="px-4 py-2"
          >
            GP
          </th>
          <th
            class="px-4 py-2"
          >
            W
          </th>
          <th
            class="px-4 py-2"
          >
            L
          </th>
          <th
            class="px-4 py-2"
          >
            PTS
          </th>
          <th
            class="px-4 py-2"
          >
            OT
          </th>
          <th
            class="px-4 py-2"
          >
            GPG
          </th>
          <th
            class="px-4 py-2"
          >
            GAPG
          </th>
          <th
            class="px-4 py-2"
          >
            SV%
          </th>
          <th
            class="px-4 py-2"
          >
            S
          </th>
          <th
            class="px-4 py-2"
          >
            SA
          </th>
          <th
            class="px-4 py-2"
          >
            S%
          </th>
          <th
            class="px-4 py-2"
          >
            FOW%
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            20
          </td>
          <td
            class="border px-4 py-2"
          >
            20
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class="league-rankings-table table-auto w-full"
    >
      <caption>
         Regular Season Rankings 
      </caption>
      <thead>
        <tr>
          <th
            class="px-4 py-2"
          >
            Overall
          </th>
          <th
            class="px-4 py-2"
          >
            GPG
          </th>
          <th
            class="px-4 py-2"
          >
            GAPG
          </th>
          <th
            class="px-4 py-2"
          >
            SV%
          </th>
          <th
            class="px-4 py-2"
          >
            S
          </th>
          <th
            class="px-4 py-2"
          >
            SA
          </th>
          <th
            class="px-4 py-2"
          >
            S%
          </th>
          <th
            class="px-4 py-2"
          >
            FOW%
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
          <td
            class="border px-4 py-2"
          >
            1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
